### PR TITLE
[INJICERT-657] Added condition for rsa and ed key generation

### PR DESCRIPTION
**Pull request by Piyush7034**

### CODE REVIEW

File: certify-service/src/main/java/io/mosip/certify/config/AppConfig.java [hunk 1]
```diff
@@ -85,11 +88,13 @@ public void run(ApplicationArguments args) throws Exception {
...
-        rsaKeyRequest.setReferenceId(Constants.EMPTY_REF_ID);
-        rsaKeyRequest.setForce(false);
-        keymanagerService.generateMasterKey("certificate", rsaKeyRequest);
+        if(pluginMode.equals("DataProvider")) {
+            KeyPairGenerateRequestDto rsaKeyRequest = new KeyPairGenerateRequestDto();
+            rsaKeyRequest.setApplicationId(Constants.CERTIFY_VC_SIGN_RSA);
+            rsaKeyRequest.setReferenceId(Constants.EMPTY_REF_ID);
+            rsaKeyRequest.setForce(false);
+            keymanagerService.generateMasterKey("certificate", rsaKeyRequest);
+        }
```
Comment (vishwa-vyom) [inline certify-service/src/main/java/io/mosip/certify/config/AppConfig.java:97]: can we not move this code also to below part of same if condition ?
I believe this key and ed key are created freshly in this release.

File: certify-service/src/main/java/io/mosip/certify/config/AppConfig.java [hunk 2]
```diff
@@ -106,16 +111,18 @@ public void run(ApplicationArguments args) throws Exception {
...
-        storeKey.setReferenceId(Constants.EMPTY_REF_ID);
-        keymanagerService.generateMasterKey("certificate", storeKey);
-        // 2. Generate an Ed25519 key later
-        KeyPairGenerateRequestDto ed25519Req = new KeyPairGenerateRequestDto();
-        ed25519Req.setApplicationId(Constants.CERTIFY_VC_SIGN_ED25519);
-        ed25519Req.setReferenceId(Constants.ED25519_REF_ID);
-        keymanagerService.generateECSignKey("certificate", ed25519Req);
+        if(pluginMode.equals("DataProvider")) {
+            // 1. Generate a master key first to enable Keymanager to store the key.
+            KeyPairGenerateRequestDto storeKey = new KeyPairGenerateRequestDto();
```
Comment (vishwa-vyom) [inline certify-service/src/main/java/io/mosip/certify/config/AppConfig.java:116]: Can we add logs like above and then say the CERTIFY_VC_SIGN_RSA KEY CHECK with equals and the same for CERTIFY_VC_SIGN_ED25519 KEY CHECK etc..

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: certify-service/src/main/java/io/mosip/certify/config/AppConfig.java (hunk 2 above)
Comment (Piyush7034) [reply certify-service/src/main/java/io/mosip/certify/config/AppConfig.java:117]: Added above logs

File: certify-service/src/main/java/io/mosip/certify/config/AppConfig.java (hunk 1 above)
Comment (Piyush7034) [reply certify-service/src/main/java/io/mosip/certify/config/AppConfig.java]: Moved RSA key generation to the condition below

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**